src/main/java/Storage.java

```diff
@@ -0,0 +1,144 @@
+import java.util.Scanner;  // Import the Scanner class
+import java.io.File; // Import the File class
+import java.io.FileWriter; // Import the FileWriter class
+import java.io.BufferedWriter; // Import the BufferedWriter class
+import java.io.IOException; // Import the IOException class to handle errors
+import java.util.ArrayList; // Import the ArrayList class
+
+public class Storage {
+
+    //file path
+    protected String filePath;
+
+    //constructor that takes in file path and creates a new file if it does not exist
+    public Storage(String filePath){
+        //set the file path
+        this.filePath = filePath;
+        //check if there is a file to read from
+        try{
+            File file = new File(filePath);
+            Scanner fileScan = new Scanner(file);
+            fileScan.close();
+        //otherwise create a new file
+        } catch(Exception e){
+            //Create a new file and directory
+            try{
+                File file = new File(filePath);
+                file.getParentFile().mkdirs();
+                file.createNewFile();
+            }catch(Exception e1){
+                e1.printStackTrace();
+            }
+        }
+    }
+   
+    //method that tand reads the file and returns an arraylist of tasks 
+    public static ArrayList<Task> readFile(String filePath){
+        //new arraylist of tasks
+        ArrayList<Task> tasks = new ArrayList<Task>();
+  
+        //File object
+        File file = new File(filePath);
+
+        //try to read the file
+        try{
+            //Create a scanner
+            Scanner fileScan = new Scanner(file);
+            //Read the file
+            while(fileScan.hasNextLine()){
+                //Read the line
+                String line = fileScan.nextLine();
+                //Split the line
+                String[] lineSplit = line.split(" \\| ");
+                //if the task is a todo
+                if(lineSplit[0].equals("T")){
```

Comment:
Magic string here too.

There are also other magic strings/numbers/literals in this file and other files. Try to use named constants or enum (if applicable) for them.